components/core/routes/Directory.py

```diff
@@ -0,0 +1,131 @@
+from flask import json
+from flask import request, Response, g
+from utils.token_utils import token_validator
+from utils.entity_utils import entity_type
+
+
+def fetch_entity_children(id):
+    token = token_validator(request.headers.get('token'))
+    if token is None:
+        return '{"error":"token error"}', 403
+    try:
+        entity = entity_type('fr')
+        fetch_response = entity.get_all(id)
+        if not isinstance(fetch_response, str):
+            return Response(response=json.dumps(fetch_response), status=200)
+        else:
+            resp = Response('{"error":"' + fetch_response + '"}', status=500)
+    except Exception as e:
+        resp = Response('{"error":"' + str(e) + '"}', status=400)
+    resp.headers['token'] = token
+    resp.headers['Access-Control-Expose-Headers'] = 'token'
+    return resp
+
+
+def fetch_entity(id):
+    token = token_validator(request.headers.get('token'))
+    if token is None:
+        return '{"error":"token error"}', 403
+    try:
+        type = request.args.get('type')
```

Comment:
Rename `type` to something else as it's a python builtin. https://docs.python.org/3.5/library/functions.html#type

Reply:
done